chainstate/broadcast_providers.go

```diff
@@ -73,6 +76,10 @@ func incorrectTxIDReturnedErr(actualTxID, expectedTxID string) error {
 	return fmt.Errorf("returned tx id [%s] does not match given tx id [%s]", actualTxID, expectedTxID)
 }
 
+func emptyBroadcastResponseErr(txID string) error {
```

Comment:
Why do we need such a distinction?

Reply:
Because if response is Nil, you can not retrieve resp.Results.TxID from it, you'll get an error.
And I'd say that there is 2 different errors: 1st - we get sth from mAPI but with a unexpected data and 2nd - we get an empty response without any data at all

Reply:
:+1: